### PR TITLE
WIP: Hybrid height support

### PR DESCRIPTION
**Pull request by corinnebosley**
Enables loading and saving (eventually) of hybrid height derived coords.

### CODE REVIEW

**Comment by corinnebosley** [comment]
First commit provides support for loading of hybrid height levels from grib2 files and includes a draft test for presence of hybrid height factory after cube has been loaded.

This test can be removed once integration tests and better unit tests have been added to iris-grib.

**Comment by corinnebosley** [comment]
@pp-mo Tests are finally passing now.  You want to take a look at the diff?

File: iris_grib/_load_convert.py [hunk 1]
```diff
@@ -1343,21 +1355,39 @@ def hybrid_factories(section, metadata):
...
             # Create the level pressure scalar coordinate.
             pv = section['pv']
             offset = scaledValue
-            coord = DimCoord(pv[offset], long_name='level_pressure',
-                             units='Pa')
+            coord = DimCoord(pv[offset], long_name=level_value_name,
+                             units=level_value_units)
             metadata['aux_coords_and_dims'].append((coord, None))
             # Create the sigma scalar coordinate.
             offset += NV // 2
```
Comment (pp-mo) [inline iris_grib/_load_convert.py]: Probably clearer/safer to replace the "+=" with a standalone calculation, i.e. `= scaledValue + NV // 2`

File: iris_grib/tests/unit/load_convert/test_vertical_coords.py [hunk 2]
```diff
@@ -173,5 +174,20 @@ def test_height_with_second_fixed_surface(self):
...
+        """
+        Temporary test for development.  Replace with proper test  or tidy up
+        for public consumption before making final PR.
+        """
+        import os.path
+        here_dirpath = os.path.dirname(__file__)
+        filepath = here_dirpath + '/../../../test_data/'
+        fname = 'faked_sample_hh_grib_data.grib2'
+        cube = load_cube(filepath + fname, 'air_temperature')
+        print(cube)
```
Comment (pp-mo) [inline iris_grib/tests/unit/load_convert/test_vertical_coords.py]: Replace print statements with an actual test.
I think it's enough to just check it *has* a factory of the expected type.
Suggest : `self.assertIsInstance(cube.aux_factories[0], HybridHeightFactory)` 
( and import the factory class somewhere above )

File: iris_grib/tests/unit/load_convert/test_vertical_coords.py [hunk 3]
```diff
@@ -173,5 +174,20 @@ def test_height_with_second_fixed_surface(self):
         self.assertEqual(metadata, expected)
 
 
+class TestRealData(tests.IrisGribTest):
+    def test_hybrid_height(self):
```
Comment (pp-mo) [inline iris_grib/tests/unit/load_convert/test_vertical_coords.py]: Being picky, call it "test_load_hybrid_height" ?

**Comment by corinnebosley** [comment]
@pp-mo Ping!

**Comment by pp-mo** [comment]
:+1: :smile:

**Pull request merged**